From f5dca9a4f5d5000958ecdf66213d1a7932d9d99d Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 22 Apr 2026 11:52:55 +0100
Subject: [PATCH 07/8] add activity animation performance override toggle

add a toggle for persist.sys.activity_anim_perf_override which
replaces stock AOSP XML-based activity transition animations with
lightweight programmatic equivalents.  fixes lag in Settings and
during app open/close on low-end MTK devices running android 16.
---
 app/src/main/java/me/phh/treble/app/Misc.kt         | 4 ++++
 app/src/main/java/me/phh/treble/app/MiscSettings.kt | 1 +
 app/src/main/res/xml/pref_misc.xml                  | 5 +++++
 3 files changed, 10 insertions(+)

diff --git a/app/src/main/java/me/phh/treble/app/Misc.kt b/app/src/main/java/me/phh/treble/app/Misc.kt
index f0193a6..1966ce1 100644
--- a/app/src/main/java/me/phh/treble/app/Misc.kt
+++ b/app/src/main/java/me/phh/treble/app/Misc.kt
@@ -406,6 +406,10 @@ object Misc: EntryStartup {
                 val value = sp.getBoolean(key, false)
                 SystemProperties.set("persist.sys.phh.securize", if (value) "1" else "0")
             }
+            MiscSettings.activityAnimPerfOverride -> {
+                val value = sp.getBoolean(key, false)
+                SystemProperties.set("persist.sys.activity_anim_perf_override", if (value) "true" else "false")
+            }
             MiscSettings.secureAdb -> {
                 val value = sp.getBoolean(key, false)
                 SystemProperties.set("persist.sys.phh.adb_secure", if (value) "1" else "0")
diff --git a/app/src/main/java/me/phh/treble/app/MiscSettings.kt b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
index 68187a3..a5eddbe 100644
--- a/app/src/main/java/me/phh/treble/app/MiscSettings.kt
+++ b/app/src/main/java/me/phh/treble/app/MiscSettings.kt
@@ -21,6 +21,7 @@ object MiscSettings : Settings {
     val disableButtonsBacklight = "key_misc_disable_buttons_backlight"
     val forceNavbarOff = "key_misc_force_navbar_off"
     val bluetooth = "key_misc_bluetooth"
+    val activityAnimPerfOverride = "key_misc_activity_anim_perf_override"
     val securize = "key_misc_securize"
     val secureAdb = "key_misc_secure_adb"
     val removeTelephony = "key_misc_removetelephony"
diff --git a/app/src/main/res/xml/pref_misc.xml b/app/src/main/res/xml/pref_misc.xml
index f7806e3..97c271f 100644
--- a/app/src/main/res/xml/pref_misc.xml
+++ b/app/src/main/res/xml/pref_misc.xml
@@ -275,6 +275,11 @@
             android:key="key_misc_securize"
             android:title="Enable Securize"
             android:summary="Disabling may fix play services on uncertified devices" />
+        <SwitchPreference
+            android:defaultValue="false"
+            android:key="key_misc_activity_anim_perf_override"
+            android:title="Activity animation performance override"
+            android:summary="Use lightweight programmatic activity transitions instead of XML animations\nFixes lag in Settings on low-end MTK devices" />
         <SwitchPreference
             android:defaultValue="false"
             android:key="key_misc_secure_adb"
-- 
2.53.0

